src/Modules/OrchardCore.Commerce/Drivers/RegionSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,91 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Commerce.AddressDataType;
+using OrchardCore.Commerce.Extensions;
+using OrchardCore.Commerce.Models;
+using OrchardCore.Commerce.ViewModels;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Settings;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Drivers;
+
+public class RegionSettingsDisplayDriver : SectionDisplayDriver<ISite, RegionSettings>
+{
+    public const string GroupId = "Region";
+    private readonly IShellHost _shellHost;
+    private readonly ShellSettings _shellSettings;
+    private readonly IHttpContextAccessor _hca;
+    private readonly IAuthorizationService _authorizationService;
+
+    public RegionSettingsDisplayDriver(
+        IShellHost shellHost,
+        ShellSettings shellSettings,
+        IHttpContextAccessor hca,
+        IAuthorizationService authorizationService,
+        IDataProtectionProvider dataProtectionProvider,
+        ILogger<RegionSettingsDisplayDriver> logger)
+    {
+        _shellHost = shellHost;
+        _shellSettings = shellSettings;
+        _hca = hca;
+        _authorizationService = authorizationService;
+    }
+
+    public override async Task<IDisplayResult> EditAsync(RegionSettings section, BuildEditorContext context)
+    {
+        var user = _hca.HttpContext?.User;
+
+        if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageRegionSettings))
+        {
+            return null;
+        }
+
+        return Initialize<RegionSettingsViewModel>("RegionSettings_Edit", model =>
+        {
+            model.AllowedRegions = section.AllowedRegions;
+
+            model.Regions = Regions.All.AsEnumerable().CreateSelectListOptions();
+        })
+            .Location("Content")
+            .OnGroup(GroupId);
+    }
+
+    public override async Task<IDisplayResult> UpdateAsync(RegionSettings section, BuildEditorContext context)
+    {
+        var user = _hca.HttpContext?.User;
+
+        if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageRegionSettings))
+        {
+            return null;
```

Comment:
Shouldn't this explicitly throw instead of return `null`? I think you'd get an exception either way, but that would be better for any future bug tracking in AI.

Reply:
I copied this from `StripeApiSettingsDisplayDriver.cs` should I change it there too?

Reply:
It was also `null` [in OC](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.ReCaptcha/Drivers/ReCaptchaSectionDisplayDriver.cs#L38-L41), that's why I used `null` here too.

Reply:
And if I should throw, then what? `UnauthorizedAccessException`?

Reply:
If it's the same in OC then it can say.